data/transactions/logic/fields.go

```diff
@@ -500,32 +578,44 @@ const (
 // AssetHoldingFieldNames are arguments to the 'asset_holding_get' opcode
 var AssetHoldingFieldNames []string
 
-// AssetHoldingFieldTypes is StackUint64 StackBytes in parallel with AssetHoldingFieldNames
-var AssetHoldingFieldTypes []StackType
-
 type assetHoldingFieldSpec struct {
 	field   AssetHoldingField
 	ftype   StackType
 	version uint64
 }
 
+func (fs *assetHoldingFieldSpec) Type() StackType {
```

Comment:
I would really like to figure out a way to shrink all the field stuff down.  Because each field is a different type, Go makes it quite hard to parameterize over them. But in many cases, calling code really only wants a map[string]uint64 and a reverse map map[uint64]string.